mysql/database.go

```diff
@@ -201,6 +201,9 @@ func (d *database) Ping() error {
 // Close terminates the current database session.
 func (d *database) Close() error {
 	if d.session != nil {
+		if d.tx != nil && !d.tx.Done() {
```

Comment:
The `d.tx.Done()` check will prevent running this rollback after a commit, it does not have any effect though.